packages/ui/src/components/Visualization/Canvas/FormV2/fields/AutoField.tsx

```diff
@@ -2,11 +2,26 @@ import { FunctionComponent, useContext } from 'react';
 import { FormComponentFactoryContext } from '../providers/FormComponentFactoryProvider';
 import { SchemaContext } from '../providers/SchemaProvider';
 import { FieldProps } from '../typings';
+import { isDefined } from '../../../../../utils';
+import { CanvasFormTabsContext } from '../../../../../providers/canvas-form-tabs.provider';
+import { useFieldValue } from '../hooks/field-value';
 
-export const AutoField: FunctionComponent<FieldProps> = ({ propName, required, onRemove }) => {
+export const AutoField: FunctionComponent<FieldProps> = ({ propName, required = false, onRemove }) => {
+  const { selectedTab } = useContext(CanvasFormTabsContext) ?? { selectedTab: 'Required' };
   const { schema } = useContext(SchemaContext);
+  const { value } = useFieldValue<object>(propName);
   const formComponentFactory = useContext(FormComponentFactoryContext);
 
+  const isFieldDefined = isDefined(value);
+  const isObjectTypefield = schema.type === 'object' || 'oneOf' in schema || 'anyOf' in schema;
+  const shouldLoadField = required || (isFieldDefined && isObjectTypefield);
+
+  if (selectedTab === 'Required' && !shouldLoadField) {
+    return null;
+  } else if (selectedTab === 'Modified' && !isFieldDefined) {
+    return null;
+  }
```

Comment:
An alert from Sonar, perhaps we could combine both if clauses into one

![image](https://github.com/user-attachments/assets/5ac10d15-4d24-46ab-a628-eaece2dcf80b)